### PR TITLE
Compose Task Cleanup

### PR DESCRIPTION
**Pull request by lancetipton**
**Ticket**: [ZEN-463](https://jira.simpleviewtools.com/browse/ZEN-463)

[Semver-Status] minor

## Context

* These changes are part of a bigger ticket to update how docker images are managed. 
* These changes were extracted to to allow for a smaller PR. Because of this; some functionality may not be used in this PR
* To allow better management of docker images, we need to improve how images are tagged

## Goal

* Fix some bugs and clean up some code
* Improve docker compose functionality
  * Add more command options

## Updates

* `repos/keg-cli/containers/*`
  * Added KEG_NODE_VERSION to build args
* `repos/keg-cli/src/tasks/docker/compose/up.js`
  * Cleaned up code, and added `recreate` option
* `repos/keg-cli/src/tasks/docker/exec.js`
  * Fixed a bug
* `repos/keg-cli/src/utils/builders/buildBaseImg.js`
  * Rename var to better describe what it does
  * Added a default tag env when building the base image
* `repos/keg-cli/src/utils/builders/buildContextEnvs.js`
  * Fixed a bug with KEG_COPY_LOCAL
* `repos/keg-cli/src/utils/docker/compose/buildComposeCmd.js`
  * Refactored the compose build command to work with image tag updates
  * Added more options to compose commands
* `repos/keg-cli/src/utils/error/throwComposeFailed.js`
  * Added better error handling when compose up task fails
* `repos/keg-cli/src/utils/docker/tags/*`
  * Added a number of build tag helpers to parse images tags dynamically
* `repos/keg-cli/src/utils/error/throwComposeFailed.js`
  * Added error handler for when a compose command fails
* `repos/keg-cli/src/utils/getters/getTagName.js`
  * Added helper for getting and images tag name
* `repos/keg-cli/src/utils/task/convertParamsToEnvs.js `
  * Updated how params are converted to envs
  * This will be used more in future PRs
  * Added better handling of param ENVs

## Testing

* Do the tests from [this PR](https://github.com/simpleviewinc/keg-hub/pull/91)
  * These code changes are used in that PRs tests

### CODE REVIEW

File: repos/keg-cli/src/tasks/tap/run.js [hunk 1]
```diff
@@ -63,8 +61,8 @@ module.exports = {
       },
       satisfy: {
         alias: [ 'sat', 'ensure' ],
-        description: 'Will check if required images are built, and build them in necessary.',
-        example: "keg ${ task } ${ action } --satisfy false",
+        description: 'Will check if required docker images are pulled and built. Will then pull and build images as needed',
+        example: `keg tap run --no-satisfy`,
```
Comment (daniel112) [inline repos/keg-cli/src/tasks/tap/run.js:65]: does adding `--no-` on the option arg make the option value falsy by default?

Comment (daniel112) [inline repos/keg-cli/src/tasks/tap/run.js:65]: did not know adding a `no-` before an option gives it a falsy value by default.
thats cool

**Review by daniel112 — APPROVED** [review]

**Review by mpcarolin — APPROVED** [review]
Looks good

**Comment by lancetipton** [comment]
Keeping open for now until all other Image from updates are merged into this branch.

File: repos/keg-cli/src/tasks/tap/run.js (hunk 1 above)
Comment (lancetipton) [reply repos/keg-cli/src/tasks/tap/run.js:65]: word

**Comment by lancetipton** [comment]
Tracking these changes in PR #99

**Pull request closed without merging**